packages/material-ui/src/Slider/Slider.js

```diff
@@ -1,1035 +1,312 @@
 import * as React from 'react';
-import PropTypes from 'prop-types';
-import clsx from 'clsx';
-import { chainPropTypes } from '@material-ui/utils';
-import withStyles from '../styles/withStyles';
+import { css, cx } from 'emotion';
+import styled from '@emotion/styled';
+import isPropValid from '@emotion/is-prop-valid';
+import { getThemeProps, useThemeVariants, propsToClassKey } from '@material-ui/styles';
 import useTheme from '../styles/useTheme';
 import { fade, lighten, darken } from '../styles/colorManipulator';
-import useIsFocusVisible from '../utils/useIsFocusVisible';
-import useEnhancedEffect from '../utils/useEnhancedEffect';
-import ownerDocument from '../utils/ownerDocument';
-import useEventCallback from '../utils/useEventCallback';
-import useForkRef from '../utils/useForkRef';
-import capitalize from '../utils/capitalize';
-import useControlled from '../utils/useControlled';
+import SliderBase from './SliderBase';
 import ValueLabel from './ValueLabel';
+import defaultTheme from '../styles/defaultTheme';
 
-function asc(a, b) {
-  return a - b;
-}
-
-function clamp(value, min, max) {
-  return Math.min(Math.max(min, value), max);
-}
-
-function findClosest(values, currentValue) {
-  const { index: closestIndex } = values.reduce((acc, value, index) => {
-    const distance = Math.abs(currentValue - value);
-
-    if (acc === null || distance < acc.distance || distance === acc.distance) {
-      return {
-        distance,
-        index,
-      };
-    }
-
-    return acc;
-  }, null);
-  return closestIndex;
-}
-
-function trackFinger(event, touchId) {
-  if (touchId.current !== undefined && event.changedTouches) {
-    for (let i = 0; i < event.changedTouches.length; i += 1) {
-      const touch = event.changedTouches[i];
-      if (touch.identifier === touchId.current) {
-        return {
-          x: touch.clientX,
-          y: touch.clientY,
-        };
-      }
-    }
-
-    return false;
-  }
+const shouldForwardProp = (prop) =>
```

Comment:
What about we spread style-related props under a specific one, like `state`? This could:

1. Avoid the need to manually list the prop to spread for each component
2. Allow the usage of more states to customization